coherence-pr-review.py

```diff
@@ -0,0 +1,266 @@
+#  Let's start with SpaCy
+
+# %%
+import spacy
+
+def named_entity_recognition_spacy(text):
+    nlp = spacy.load('en_core_web_sm')
+    doc = nlp(text)
+    named_entities = [ent.text for ent in doc.ents]
+    return named_entities
+
+
+# %%
+named_entities = [named_entity_recognition_spacy(sentence) for sentence in sentences]
+print('Named Entities:', named_entities)
+
+# %%
+def specificity_ne_spacy(story):
+    sentences = tokenize_text(story)
+    print('Sentences:', sentences)
+    tagged_sentences = [pos_tagging(sentence) for sentence in sentences]
+    print('Tagged Sentences:', tagged_sentences)
+    named_entities = [named_entity_recognition_spacy(sentence) for sentence in sentences]
+    print('Named Entities:', named_entities)
+    events = [extract_events(sentence) for sentence in sentences]
+    print('Events:', events)
+
+
+# %% [markdown]
+# Testing the specifity_ne_spacy function :
+
+# %%
+specificity_ne_spacy(data.iloc[0]['Story'])
+
+# %%
+specificity_ne_spacy(data.iloc[1]['Story'])
+
+# %% [markdown]
+# # 2.Personal Context 
+
+# %%
+import nltk
+from nltk.sentiment import SentimentIntensityAnalyzer
+from gensim import corpora, models
+
+def evaluate_personal_context(text):
+    # Tokenize text into sentences
+    sentences = nltk.sent_tokenize(text)
+
+    # Sentiment Analysis
+    sia = SentimentIntensityAnalyzer()
+    sentiment_scores = [sia.polarity_scores(sentence)["compound"] for sentence in sentences]
+
+    # Topic Modeling
+    tokenized_sentences = [nltk.word_tokenize(sentence.lower()) for sentence in sentences]
+    dictionary = corpora.Dictionary(tokenized_sentences)
+    corpus = [dictionary.doc2bow(tokens) for tokens in tokenized_sentences]
+    lda_model = models.LdaModel(corpus, num_topics=3, id2word=dictionary)
+
+    # Extract most significant topics
+    topics = [lda_model.get_document_topics(doc) for doc in corpus]
+    most_significant_topics = [max(topic, key=lambda x: x[1]) for topic in topics]
+
+    # Get actual topics
+    actual_topics = [lda_model.print_topic(topic[0]) for topic in most_significant_topics]
+
+    # Return sentiment scores and actual topics
+    return sentiment_scores, actual_topics
+
+
+# %% [markdown]
+# In the context of topic modeling with LDA, the weights assigned to each word in a topic represent the importance or prevalence of that word within the topic. In the output you provided
+# 
+# Understanding the significance of individual words and their weights within a topic can help provide insights into the key themes and subjects present in the text.
+
+# %%
+def test_personal_context(text):
+    sentiment_scores, actual_topics = evaluate_personal_context(text)
+    print("Sentiment Scores:", sentiment_scores)
+    print()
+    print("Actual Topics:")
+    for topic in actual_topics:
+        print(topic)
+
+# %%
+test_personal_context(data.iloc[0]['Story'])
+
+# %%
+test_personal_context("i lost my best friend and im so happy")
```

Comment:
Good move to test some obvious examples where we have an expected return. However, it would be nice to create some of them, store the expected result within a csv file, and compare within a script actual and expected results. You can use `pytest` as a framework for such a task.